packaging/technical-addon/Splunk_TA_otel/windows_x86_64/bin/Splunk_TA_otel_utils.ps1

```diff
@@ -1,63 +1,85 @@
 $otelProcessName=$args[0]
-$splunkTAOtelLogFileName=$args[1]
-$splunkTAOtelLogDir=Split-Path -Path $splunkTAOtelLogFileName
-$splunkTAPSLogFile=Join-Path -Path $splunkTAOtelLogDir -ChildPath Splunk_TA_otelutils.log
 
 function otelLogWrite
 {
    Param ([string]$logstring)
    $timestamp = Get-Date -Format "yyyy-MM-dd HH:mm:ss.fff K"
-   Add-content $splunkTAPSLogFile -value "$timestamp $logstring"
+
+   # Write-Warning is much less verbose than Write-Error and the other Write-* may end up
+   # not being captured by the parent process (expected to be the cmd that launched the script)
+   Write-Warning "$timestamp Splunk_TA_otel_utils.ps1: $logstring"
+}
+
+function getParentProcessId($id)
+{
+	return (Get-WmiObject Win32_Process | Where-Object ProcessId -eq $id).ParentProcessId
+}
+
+function getParentAndGrandParent($id)
+{
+	try {
+		$parentId = getParentProcessId($id)
+		otelLogWrite "INFO Parent process id: $parentId"
+		$grandParentId = getParentProcessId($parentId)
+		otelLogWrite "INFO GrandParent process id: $grandParentId"		
+	}
+	catch {
+		# Use Write-Error directly here to get detailed information.
+		Write-Error "ERROR getParentAndGrandParent: $_"
+	}
+
+	return $parentId, $grandParentId
 }
 
-function isOtelProcessRunning($processName)
+function getOtelProcessPid($processName, $parentId)
 {
-	$parent = (Get-WmiObject win32_process | ? processid -eq $PID).parentprocessid
-	$Global:parentPid = $parent
-	otelLogWrite "INFO Parent process id: $parent"
-	$child = (Get-WmiObject win32_process | ? parentprocessid -eq $parent | ? processname -eq $processName).processid
-	$Global:otelPid = $child
-	otelLogWrite "INFO Collector process id: $child"
-	$otelProcess = Get-Process -Id $child
+	otelLogWrite "INFO Getting PID for $processName with parent PID $parentId ..."
+	$procPid = (Get-WmiObject Win32_Process | Where-Object ParentProcessId -eq $parentId | Where-Object ProcessName -eq $processName).ProcessId
+	otelLogWrite "INFO Collector process id: $procPid"
+	$otelProcess = Get-Process -Id $procPid -ErrorAction Ignore
 	if($otelProcess)
 	{
-		return $true
+		return $procPid
 	}
 	else
 	{
-		return $false
+		return -1
 	}
 }
 
-function waitForExit($parent) 
+function forceStopProcess($processId)
 {
-	Wait-Process -Id $parent
-	otelLogWrite "INFO Parent process exited"
+	otelLogWrite "INFO Stopping $processId if stil running"
+	# Use a pipeline to avoid an error message if the process already finished
+	Get-WmiObject Win32_Process | Where-Object ProcessId -eq $processId | ForEach-Object { taskkill.exe /f /pid $_.ProcessId }
 }
 
-function forceStopOtelProcess($processId)
-{
-	Stop-Process -Id $processId -Force
-	Start-Sleep -Seconds 1
-}
-
-function CheckOtelProcessStop($processId)
-{
-	$processInfo = Get-WmiObject win32_process | ? processid -eq $processId
-	if($processInfo) {
-		otelLogWrite "ERROR Otel agent didn't stop"
+otelLogWrite "INFO Splunk_TA_otelutils.ps1 started"
+try {
+	otelLogWrite "INFO sleeping ..."
+	start-sleep -s 3
```

Comment:
I know this is just moved and not changed with this PR, but maybe as a future enhancement we should be doing some kind of polling with a deadline, rather than a single sleep 3 seconds and check here.